tools/automation/cli_linter/rules/help_rules.py

```diff
@@ -0,0 +1,26 @@
+# --------------------------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License.txt in the project root for license information.
+# --------------------------------------------------------------------------------------------
+
+from ..rule_decorators import help_file_entry_rule
+
+
+@help_file_entry_rule('Checking unrecognized commands and command-groups in help...')
+def unrecognized_help_entry_rule(linter, help_entry):
+    if help_entry not in linter.get_commands().union(linter.get_command_groups()):
```

Comment:
Since help entry rules would often be concerned with this "command or group" scenario, perhaps it would make sense to precompile this list in the linter so you don't have to keep creating this union, testing it, and throwing it away. Maybe something like:

```Python
def find_command_or_group_help(self, name):
   """ Return the help entry for the specified command or command group, or None if not found. """
   return self._master_help.get(name, None)
```